program/CORE-V IDE prelminary project proposal.md

```diff
@@ -0,0 +1,251 @@
+3. �OpenHW IDE� splash-screen and "about" box with openHW Group graphics. 
+4. Set version to v0.0.1
+5. Build on latest, released Eclipse, CDT and GNU MCU  plug-ins (now known as Eclipse Embedded CDT (C/C++ Development Tools))
+6. Latest Embecosm CORE-V toolchain
+7. OpenOCD (Open On-Chip Debugger, with JTAG support) for CORE-V
+8. Simple Eclipse project ready-to-run on Genesys 2 board (e.g. LED toggle) including Opella-LD OpenOCD launch (Opella-LD is 	Ashling's low-cost, FTDI based probe not to be confused with the existing Opella-XD)
+9. OpenHW IDE getting started guide documentation with Genesys 2 board 
+
+
+### Future enhancements:
+
+10. Adding Linux support
+11. Automating the build to �pull� all the latest components from their respective repos etc
+12. Improved documentation and more board examples
+13. Various improvements
+
+
+
+## Explanation of why OpenHW should do this project
+
+The industry will require a Embedded CDT IDE tuned to CORE-V. 
+- OpenHW is developing the CORE-V RTL, FPGA instantiations of the CORE-V RTL, and the s/w toolchain to support CORE-V.
+- OpenHW has a close parternship with the Eclipse Foundation. 
+- Building the CDT enhancements for CORE-V builds on the strengths of the OpenHW ecosystem. 
+
+For the above reasons, OpenHW is best placed to do this needed work.
+
+## Industry landscape: description of competing, alternative, or related efforts in the industry
+
+### Related efforts to be described
+
+- Eclipse CDT 
+- (GNU MCU) Eclipse Embedded CDT (Describe support for existing RISC-V devices)
+- Eclipse Theia
+
+## External dependencies
+ 
+The first build of CORE-V CDT builds on the following:
+- Eclipse Platform 2020-09
+- Eclipse CDT 10.0
+- Other components from Eclipse SimRel 2020-09
+- A suitable tool chain (see options below)
+
+The CORE-V IDE will be capable of integration with the following tool chain options. As per the project requirements above, the main focus of the CORE-V IDE will be the OpenHW CORE-V reference toolchain.
+
+### PULP Code Base As-Is
+
+This is a research tool chain from ETHZ and U of Bologna which has the following attributes:
+- It is updated often as research projects need.
+- A precompiled version of that toolchain is available, compiled and tested from Embecosm, updated weekly.
+- Drawbacks are that it is out of date (based on earlier release of GCC), doesn't meet coding standards, and lacks testing
+
+
+
+### Standard RISC-V tool chain
+This is the standard RISC-V GCC tool chain whose attributes include:
+- The development is hosted by the FSF (Free software foundation). It is completely up to date with GCC and binutils. There are many commits per day.
+- It has no Pulp features
+- A precompiled version of that toolchain is available from Embecosm and possibly others. From Embecosm, the latest version is updated weekly. The latest table release is 10.2.0
+- It is currently used in the Ashling IDE product.

```

Comment:
For completeness, we should include

### Standard Clang/LLVM tool chain

- The development is hosted by the LLVM Foundcation. It is completely up to date with GCC and binutils. There are many commits per day.
￼- It has no Pulp features
￼- A precompiled version of that toolchain is available from Embecosm and possibly others. From Embecosm, the latest version is updated weekly. The latest stable release is 10.0.1

----